### PR TITLE
link image ID

### PR DESCRIPTION
**Pull request by will-moore**
# What this PR does

Turn ```Image ID: x``` into a link to ```webclient/?show=image-x```.

Screen Shot 2019-07-18 at 22 59 44

To test:
 - Edit description of E.g. Project, Dataset, Image etc and add a Comment, containing "Image ID: 123" and this will get turned into a link to that image.

### CODE REVIEW

**Comment by will-moore** [comment]
Travis failed with 
```
install-jdk.sh 2019-07-17
Expected feature release number in range of 9 to 14, but got: 8
[31;1mThe command "~/bin/install-jdk.sh --target "/home/travis/oraclejdk8" --workspace "/home/travis/.cache/install-jdk" --feature "8" --license "BCL"" failed and exited with 3 during .[0m
```

**Comment by joshmoore** [comment]
Pushed a commit trying out openjdk8.

**Comment by joshmoore** [comment]
Now:
```
[2BError: No such container:path: openmicroscopy_omero_1:/
```

**Comment by joshmoore** [comment]
Re-opening with fix from gh-6089

**Comment by mtbc** [comment]
Do we have an "Object ID: 1234" format elsewhere? I wonder if, say, "Object:1234" or "Object #1234" might be worth matching. Also wondering if it'd be nice to also recognize these patterns in KVP values. Just some thoughts.

Happy with this as it is too though, fine to merge if "Object ID: 1234" is the way we're going: the PR works fine for me as-is.

**Comment by will-moore** [comment]
I added support for other types, E.g Project, Dataset, Screen, Plate, Well, ROI
and for "Image: 123" or "Image ID: 123".

**Comment by manics** [comment]
How about `Object:ID` without the space, to match the format of the omero CLI ?

File: components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js [hunk 1]
```diff
@@ -62,7 +62,14 @@ OME.getURLParameter = function(key) {
 
 var linkify = function(input) {
     var regex = /(https?|ftp|file):\/\/[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]/g;
-    return input.replace(regex, "<a href='$&' target='_blank'>$&</a>");
+    input = input.replace(regex, "<a href='$&' target='_blank'>$&</a>");
+    return linkImages(input);
+};
+var linkImages = function(input) {
```
Comment (manics) [inline components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js:68]: Change the name `linkImages` to reflect the expanded scope?

File: components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js [hunk 2]
```diff
@@ -62,7 +62,14 @@ OME.getURLParameter = function(key) {
...
+    var regex = /(Project|Dataset|Image|Screen|Plate|Well|ROI)(?: ID)*:(?: )*([0-9]+)/g;
+    return input.replace(regex, function($0, $1, $2) {
+        return "<a title='Browse to " + $1 + "' href='" + WEBCLIENT.URLS.webindex + "?show=" + $1.toLowerCase() + "-" + $2 + "'>" + $0 + "</a>"
```
Comment (manics) [inline components/tools/OmeroWeb/omeroweb/webclient/static/webclient/javascript/ome.webclient.actions.js:71]: `Browse to Project` doesn't sound right, how about `Go to ...`, `Load ...` or `Open ...`?

**Review by manics — COMMENTED** [review]
Functionally works fine

**Comment by manics** [comment]
Unrelated to this PR: `/webclient/?show=roi-123` goes to the image containing the ROI, would be nice if it could open the image viewer and show the ROI

**Comment by will-moore** [comment]
Ported to https://github.com/ome/omero-web/pull/16

**Pull request closed without merging**